server-protocol.md

```diff
@@ -269,9 +268,37 @@ blob (for the VERSION message, as well as all application-provided payloads).
 The `message` response will also include `id`, copied from the `id` of the
 `add` message (and used only by the timing-diagram tool).
 
-The Rendezvous Server does not de-duplicate messages, nor does it retain
+The Mailbox server does not de-duplicate messages, nor does it retain
 ordering: clients must do both if they need to.
 
+The `close` command accepts an optional "mood" string: this allows clients to
+tell the server (in general terms) about their experiences with the wormhole
+interaction. The server records the mood in its "usage" record, so the server
+operator can get a sense of how many connections are succeeding and failing.
+The moods currently recognized by the Mailbox server are:
+
+* `happy` (default): the PAKE key-establishment worked, and the client saw at
+  least one valid encrypted message from its peer
+* `lonely`: the client gave up without hearing anything from its peer
+* `scary`: the client saw an invalid encrypted message from its peer,
+  indicating that either the wormhole code was typed in wrong, or an attacker
+  tried (and failed) to guess the code
+* `errory`: the client encountered some other error: protocol problem or
+  internal error
+
+The server will also record `pruney` if it deleted the mailbox due to
+inactivity, or `crowded` if more than two sides tried to access the mailbox.
+
+A client sending a `close` message issues a `closed` message on the other client,
+to notify it about the intended session teardown. The latter client can then
+continue to send messages (so-called "half-closed state") until sending a `close`
+message too, which in turn will send the `closed` command to the first client and
```

Comment:
Here, too, I think we should link to the "closing" ticket and come to a conclusion there..? Also we should make it clear here between "current behavior" and "proposed future change". How will a "new" client know if it is talking to an "old" client (that doesn't do this)?

(e.g. what's "a closed message"? Presumably you mean a special sort of number-phase message here? Or something else?)

Reply:
(Hmm, re-reading this and it's not clear to me who "issues a closed message": do you mean the server does this? Or the other client?)

Anyway, I think this is best concluded in the existing ticket about graceful session shutdown: https://github.com/magic-wormhole/magic-wormhole-mailbox-server/issues/31